lib/index.js

```diff
@@ -259,7 +263,12 @@ exports.tooManyRequests = function (message, data) {
 
 exports.internal = function (message, data, statusCode) {
 
-    var error = (data instanceof Error ? exports.wrap(data, statusCode, message) : exports.create(statusCode || 500, message));
+    return internals.internal(message, data, statusCode, exports.internal);
+};
+
+internals.internal = function (message, data, statusCode, ctor) {
```

Comment:
How about a more descriptive name? `generate` or `build` or something better that I can't think of.